datafusion/optimizer/src/projection_push_down.rs

```diff
@@ -332,16 +330,6 @@ fn optimize_plan(
                 }
             })?;
 
-            let new_schema = DFSchema::new_with_metadata(
-                schema
-                    .fields()
-                    .iter()
-                    .filter(|x| new_required_columns.contains(&x.qualified_column()))
-                    .cloned()
-                    .collect(),
-                schema.metadata().clone(),
-            )?;
```

Comment:
This code was problematic and is now replaced with the code used elsewhere

Reply:
yes I agree having the caller have to specify (correctly) the aggregate schema is a recipe for disaster